src/target/stm32h7.c

```diff
@@ -207,14 +213,44 @@ bool stm32h7_probe(target_s *target)
 	target->target_storage = priv_storage;
 
 	/* Build the RAM map */
-	/* Table 7. Memory map and default device memory area attributes RM0433, pg130 */
 	target_add_ram(target, 0x00000000, 0x10000); /* ITCM RAM,   64kiB */
 	target_add_ram(target, 0x20000000, 0x20000); /* DTCM RAM,  128kiB */
-	target_add_ram(target, 0x24000000, 0x80000); /* AXI RAM,   512kiB */
-	target_add_ram(target, 0x30000000, 0x20000); /* AHB SRAM1, 128kiB */
-	target_add_ram(target, 0x30020000, 0x20000); /* AHB SRAM2, 128kiB */
-	target_add_ram(target, 0x30040000, 0x08000); /* AHB SRAM3,  32kiB */
-	target_add_ram(target, 0x38000000, 0x10000); /* AHB SRAM4,  64kiB */
+	switch (target->part_id) {
+	case ID_STM32H72x: {
+		/* Table 6. Memory map and default device memory area attributes RM0468, pg133 */
+		target_add_ram(target, 0x24000000, 0x20000); /* AXI RAM, 128kiB */
+		target_add_ram(target, 0x24020000, 0x30000); /* AXI/ITCM RAM, 192kiB */
+		target_add_ram(target, 0x30000000, 0x4000);  /* AHB SRAM1, 16kiB */
+		target_add_ram(target, 0x30004000, 0x4000);  /* AHB SRAM2, 16kiB */
+		target_add_ram(target, 0x38000000, 0x4000);  /* AHB SRAM4, 16kiB */
+		break;
+	}
+	case ID_STM32H74x: {
+		/* Table 7. Memory map and default device memory area attributes RM0433, pg130 */
+		target_add_ram(target, 0x24000000, 0x80000); /* AXI RAM,   512kiB */
+		target_add_ram(target, 0x30000000, 0x20000); /* AHB SRAM1, 128kiB */
+		target_add_ram(target, 0x30020000, 0x20000); /* AHB SRAM2, 128kiB */
+		target_add_ram(target, 0x30040000, 0x08000); /* AHB SRAM3,  32kiB */
+		target_add_ram(target, 0x38000000, 0x10000); /* AHB SRAM4,  64kiB */
+		break;
+	}
+	case ID_STM32H7Bx: {
+		/* Table 6. Memory map and default device memory area attributes RM0455, pg131 */
+#if 1
+		target_add_ram(target, 0x24000000, 0x40000); /* AXI RAM1, 256kiB */
+		target_add_ram(target, 0x24040000, 0x60000); /* AXI RAM2, 384kiB */
+		target_add_ram(target, 0x240a0000, 0x60000); /* AXI RAM3, 384kiB */
+#else
+		target_add_ram(target, 0x24000000, 0x100000); /* AXI RAM123, 1024kiB [256+384+384] contiguous */
+#endif
```

Comment:
Given this is contiguous, think it's fine to just do the one `target_add_ram()`, which also saves on space a little. Same for any other contiguous ranges in the surrounding code - RAM isn't handled like Flash, so this is only telling GDB about the valid IO ranges when `inaccessible-by-default` is enabled.

Reply:
Please do, yeah